modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/components/ViolationPopover.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import './ViolationPopover.scss';
+
+import ClayIcon from '@clayui/icon';
+import ClayList from '@clayui/list';
+import ClayPopover from '@clayui/popover';
+import React, {useCallback, useMemo, useState} from 'react';
+
+import {useObserveRect} from '../hooks/useObserveRect';
+
+import type {RuleRaw} from '../hooks/useA11y';
+
+const Overlay = React.forwardRef<
+	HTMLDivElement,
+	React.ButtonHTMLAttributes<HTMLDivElement>
+>(({style, ...othersProps}, ref) => (
+	<div {...othersProps} className="a11y-overlay" ref={ref} style={style}>
+		<div className="a11y-indicator">
+			<ClayIcon symbol="info-circle" />
+		</div>
+		<div className="a11y-backdrop" />
+	</div>
+));
+
+type ViolationProps = {
+	target: string;
+	violations: Array<string>;
+	rules: Record<string, RuleRaw>;
+};
+
+export function ViolationPopover({rules, target, violations}: ViolationProps) {
+	const [visible, setVisible] = useState(false);
+	const [bounds, setBounds] = useState<React.CSSProperties>();
+
+	const node = useMemo(() => document.querySelector(target), [target]);
+
+	useObserveRect(
+		useCallback(
+			(bounds) => {
+				bounds = bounds ?? (node as Element).getBoundingClientRect();
+
+				setBounds({
+					height: bounds.height,
+					left: bounds.left,
+					top: bounds.top,
+					width: bounds.width,
+				});
+			},
+			[node]
+		),
+		node
+	);
+
+	return (
+		<ClayPopover
+			className="a11y-popover"
+			header={
+				<>
+					<div className="inline-item">
+						<ClayIcon
+							className="text-danger"
+							symbol="info-circle"
+						/>
+					</div>
+					<div className="inline-item inline-item-after">
+						<span>
+							{Liferay.Language.get('accessibility-violations')}
+						</span>
+					</div>
+				</>
+			}
+			onShowChange={setVisible}
+			show={visible}
+			trigger={<Overlay style={bounds} />}
+		>
+			<div className="list-group">
```

Comment:
Can you use our ClayList components here?

Reply:
Yeah, initially I was wanting to use the `ClayList` components but the markup is `ul` and I need it to be `div` because the items are `button`. Maybe I'll create an issue in Clay to analyze the possibility of allowing us to change the tag of components, I saw in some libraries a very common behavior for this with the `as` prop, something like `as="div"`.